### DISCUSSION TITLE
Prisma throwing error saying integer data is not integer but it is integer

### DISCUSSION THREAD

**Discussion by touhidurrr** [prisma/prisma#25946] [Q&A] [answered]
### Question

Am I doing something wrong here or it is really a bug that needs to be solved?

Sauce: #25939

**Comment by nurul3101** [comment]
Hello @touhidurrr 👋 

I am trying to understand the issue better.
Based on the logs you shared in the linked GitHub Issue:

> modelName: 'Variable',
>  message: "Conversion failed: number must be an integer in column 'createdAt', got '1735325788553.0'

The value `1735325788553.0`  is a numeric value that includes a decimal point, which typically signifies a floating-point (REAL) number, so this won't be considered as an Integer, right?

So, shouldn't this error be valid? Could you try inserting without the decimal and check?

**Reply by touhidurrr** [reply]
@nurul3101 no. I already shared my schema and db SQL output. So, `typeof(createdAt)` returns integer which means the database is surely not storing the data as real or text or numeric type.
Also `select createdAt from Variable` also does not display any `.0` part. So, I am not sure where Prisma got that `.0` part from and so my best guess would be that it is some kind of internal conversion issue. Unless someone notices something that I might not be aware of.

**Reply by ludralph** [reply, marked as answer]
Hi @touhidurrr 

I have gone through the issue you created and can confirm that `createdAt` is returned as a float instead of an integer. I think it's a bug and will add the bug label to that issue so that the engineers can take a deeper look.
```
 Conversion failed: number must be an integer in column 'createdAt', got '1738834045465.0'
```
